front/src/modules/infra/components/InfraSelector/InfraSelectorGrantsManager.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useMemo } from 'react';
+
+import { Select } from '@osrd-project/ui-core';
+import { ChevronDown, ChevronUp } from '@osrd-project/ui-icons';
+import { useTranslation } from 'react-i18next';
+
+import { mockedEditoastApi, type GetSubjectsResponse } from 'common/api/mock/mockEditoastApi';
+import { DEFAULT_GRANT, GrantsLabel } from 'modules/infra/consts';
+import type { GrantsLabelKeys } from 'modules/infra/type';
+import generateGrantSelectProps from 'modules/infra/utils/generateGrantSelectProps';
+import { capitalizeFirstLetter } from 'utils/strings';
+
+type InfraSelectorGrantsManagerProps = {
+  infraId: number;
+  userGrant: string;
+  resourceGrants?: {
+    [grant: string]: string[];
+  };
+  subjectList?: GetSubjectsResponse;
+};
+
+export default function InfraSelectorGrantsManager({
+  infraId,
+  userGrant,
+  resourceGrants = {},
+  subjectList = [],
+}: InfraSelectorGrantsManagerProps) {
+  const [displayGrantSection, setDisplayGrantSection] = React.useState(false);
+  const { t } = useTranslation('infraManagement');
+
+  const usersList = useMemo(
+    () => subjectList.filter((subject) => subject.type === 'user' && subject.id !== 1),
+    [subjectList]
+  );
+
+  const [postUsersGrantsByResourceId, { data: usersInfraGrant }] =
+    mockedEditoastApi.endpoints.postUsersGrantsByResourceId.useMutation();
+
+  useEffect(() => {
+    const getUserGrantsByInfraId = async () => {
+      const subjectsIdList = subjectList.map((subject) => subject.id);
+      try {
+        await postUsersGrantsByResourceId({
+          subjects_id: subjectsIdList,
+          resource_type: 'infra',
+          resource_id: infraId,
+        });
```

Comment:
If I understand correctly, this endpoint is used to fetch all the grants of the subjects list (users ids) for a specific infra. Would it be possible to have the same endpoint which will return the grant of the users on the infras at once ?